packages/core/src/timeline.ts

```diff
@@ -76,6 +77,15 @@ export class Timeline {
       }
     }
 
+    Inspector.trace({
+      timestamp: new Date().toISOString(),
+      stage: 'delivered',
```

Comment:
Are the stage values an enum? do they show up differently in the inspector depending on the value here?